### PR TITLE
implemented expm multiply

### PR DESCRIPTION
**Pull request by kaihsin**
This update include

1. onenormest(A,p), which estimate |A^p|_1  : one norm of a matrix A to the power of p
2. expm_multiply() which is krylov-subspace-free expmv.
3. [Note] This is branch out from khwu/expr_refactor

### CODE REVIEW

File: lib/BloqadeExpr/src/Lowlevel/Lowlevel.jl [hunk 1]
```diff
@@ -9,12 +9,17 @@ using Preferences
 using Adapt
 using LaTeXStrings
 using LinearAlgebra
+using ForwardDiff
```
[the remark this answers is not in the document]
Comment (kaihsin) [reply lib/BloqadeExpr/src/Lowlevel/Lowlevel.jl:12]: Yes, this PR includes derivative of Hamiltonian struct.

File: lib/BloqadeExpr/src/Lowlevel/linalg.jl [hunk 2]
```diff
@@ -77,3 +96,114 @@ end
...
+end
+
+function add_I(A::SumOfLinop, c::Number)
+    Iop = LinearAlgebra.I(size(A,1))
+
+    if nthreads() > 1
+        return SumOfLinop{RegularLinop}((A.fvals...,c),(A.ts...,ThreadedMatrix(Iop)))
+    else
+        return SumOfLinop{RegularLinop}((A.fvals...,c),(A.ts...,Iop))
+    end
```
[the remark this answers is not in the document]
Comment (kaihsin) [reply lib/BloqadeExpr/src/Lowlevel/linalg.jl:135]: I basically follow the code done in lower.jl

File: lib/BloqadeExpr/Project.toml [hunk 3]
```diff
@@ -6,6 +6,7 @@ version = "0.1.14"
 Adapt = "79e6a3ab-5dfb-504d-930d-738a2a938a0e"
 BitBasis = "50ba71b6-fa0f-514d-ae9a-0916efc90dcf"
 BloqadeLattices = "bd27d05e-4ce1-5e79-84dd-c5d7d508bbe4"
+ForwardDiff = "f6369f11-7733-5829-9624-2563aa707210"
```
[the remark this answers is not in the document]
Comment (kaihsin) [reply lib/BloqadeExpr/Project.toml:9]: done

**Review by weinbe58 — APPROVED** [review]

**Comment by weinbe58** [comment]
Need to improve coverage for `expm_multiply` implementation.

**Pull request merged**